torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -24,6 +24,25 @@ LoopNestGenerator::LoopNestGenerator(
   generate(exprs);
 }
 
+namespace {
+// Currently, allocation of smem tennsors and indexing is
+// broken when computeAt axes are thread-parallelized. This check
+// throws an exception if such tensors are detected.
+// TODO: Fix the allocation and indexing of such tensors.
+void failIfUnsupported(TensorView* tv) {
+  for (size_t i = 0; i < tv->getThisComputeAtAxis(); i++) {
+    IterDomain* compute_at_dim = tv->getComputeAtAxis(i).first;
+    const auto memory_type = tv->getMemoryType();
+    if (memory_type == MemoryType::Shared && compute_at_dim->isThreadDim()) {
+      std::stringstream ss;
```

Comment:
why not build the error msg directly with `TORCH_INTERNAL_ASSERT` ?

Reply:
Printing `TensorView*` with `TORCH_INTERNAL_ASSERT` didn't work for me (just its address was printed). Using `std::stringstream` was a workaround.